### PR TITLE
Split CNI and updated for 1.23

### PR DESCRIPTION
**Pull request by kschumy**
*Issue #, if available:*

#### Description of changes:
_TL;DR: Split CNI by release branches and updated 1.23's version_

* **For 1-20, 1-21, and 1-22**
  * Moved everything branch-specific that was originally in the project's directory into release branch directories. Specifically, this was `ATTRIBUTION.txt`, `CHECKSUMS`, `GIT_TAG`, `go.mod`, `go.sum`, and `patches/`
  * Added `GOLANG_VERSION`
* **For 1-23**
    * Added `GIT_TAG` with new version ([v.1.1.1](https://github.com/containernetworking/plugins/tree/v1.1.1)). See [changelog](https://github.com/containernetworking/plugins/releases/tag/v1.1.1)
    * Added `GOLANG_VERSION` with version ([v.1.17](https://github.com/containernetworking/plugins/blob/v1.1.1/scripts/release.sh#L19))
    * Did NOT add patches because they were no longer needed
      * `0001-Update-libcni-to-0.8.1.patch` -- upstream [PR](https://github.com/containernetworking/plugins/pull/580) was merged, and [commit](https://github.com/containernetworking/plugins/commits/v1.1.1?after=4744ec27b89c083194e7df498de50f03a8a1d3ec+56&branch=v1.1.1&qualified_name=refs%2Ftags%2Fv1.1.1) is included in this version of CNI
      * `0002-go.mod-github.com-buger-jsonparser-v1.1.1.patch` -- upstream [PR](https://github.com/containernetworking/plugins/pull/603) was merged, and [commits](https://github.com/containernetworking/plugins/commits/v1.1.1?after=4744ec27b89c083194e7df498de50f03a8a1d3ec+109&branch=v1.1.1&qualified_name=refs%2Ftags%2Fv1.1.1) (all the "go.mod: ..." ones) are included in this version of CNI

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kschumy** [comment]
I did not include release branches for 1-18 and 1-19 because they're no longer supported. If people feel differently about this, let me know.

**Comment by kschumy** [comment]
/unhold

job changes are done

**Comment by kschumy** [comment]
I'll update checksums when 1.23 fails
